jmclient/jmclient/client_protocol.py

```diff
@@ -283,52 +293,48 @@ def unconfirm_callback(self, txd, txid):
             if v["txd"]["outs"] == txd["outs"]:
                 offerinfo = v
                 break
-        if not offerinfo:
-            jlog.info("Failed to find notified unconfirmed transaction: " + txid)
-            return
-        removed_utxos = self.client.wallet.remove_old_utxos(txd)
-        jlog.info('saw tx on network, removed_utxos=\n{}'.format('\n'.join(
-            '{} - {}'.format(u, fmt_tx_data(tx_data, self.client.wallet))
-            for u, tx_data in removed_utxos.items())))
+        if offerinfo is None:
+            return False
         to_cancel, to_announce = self.client.on_tx_unconfirmed(offerinfo,
-                                                               txid, removed_utxos)
+                                                               txid)
         self.client.modify_orders(to_cancel, to_announce)
+
+        txinfo = tuple((x["script"], x["value"]) for x in txd["outs"])
+        confirm_timeout_sec = float(jm_single().config.get(
+            "TIMEOUT", "confirm_timeout_hours")) * 3600
+        task.deferLater(reactor, confirm_timeout_sec,
+                        self.client.ws.check_callback_called,
+                        txinfo, self.confirm_callback, "confirmed",
+        "transaction with outputs " + str(txinfo) + " not confirmed.")
+
         d = self.callRemote(commands.JMAnnounceOffers,
                             to_announce=json.dumps(to_announce),
                             to_cancel=json.dumps(to_cancel),
                             offerlist=json.dumps(self.client.offerlist))
         self.defaultCallbacks(d)
+        return True
 
-    def confirm_callback(self, txd, txid, confirmations):
+    def confirm_callback(self, txd, txid, confirms):
```

Comment:
this whole code is mostly duplicating `unconfirm_callback`

Reply:
Refactored the transaction matching to be common ( https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/359/commits/eac7fd544b6728cf1753d935927f2148f738c32d ) , but left the other (callRemote) in place, seems pretty minor.